sql/database.go

```diff
@@ -277,13 +297,32 @@ func Version(uri string) (int, error) {
 	if err != nil {
 		return 0, fmt.Errorf("open db %s: %w", uri, err)
 	}
-	db := &Database{pool: pool}
+	db := &sqliteDatabase{pool: pool}
 	defer db.Close()
 	return version(db)
 }
 
-// Database is an instance of sqlite database.
-type Database struct {
+// Database represents a database.
+type Database interface {
+	Executor
+	QueryCache
+	Close() error
+	QueryCount() int
+	QueryCache() QueryCache
+	Tx(ctx context.Context) (Transaction, error)
+	WithTx(ctx context.Context, exec func(Transaction) error) error
+	TxImmediate(ctx context.Context) (Transaction, error)
+	WithTxImmediate(ctx context.Context, exec func(Transaction) error) error
+}
+
+// Transaction represents a transaction.
+type Transaction interface {
+	Executor
+	Commit() error
+	Release() error
```

Comment:
`Release()` is basically "rollback if not committed" in addition to releasing the connection, so it kind of also belongs to transaction just as well as `Commit`. Basically we have "Commit - Release" pair hear instead of "Commit - Rollback"